src/vecgeom/VecgeomTrackView.i.hh

```diff
@@ -191,9 +213,10 @@ CELER_FUNCTION void VecgeomTrackView::move_internal(real_type dist)
     CELER_EXPECT(this->has_next_step());
     CELER_EXPECT(dist > 0 && dist < next_step_);
 
-    // Move and update next_step_
+    // Move and update next_step_ and safety_
     axpy(dist, dir_, &pos_);
     next_step_ -= dist;
+    safety_ = this->safety(pos_);
```

Comment:
Even though I think it's fine to allocate the extra space for safety, I don't think we want to *always* recalculate it at every internal movement. Could you change this so it's lazily calculated like next_step?

Reply:
It only makes sense to update the safety if it's being used. For photons we'll never need the safety, for example. If `move_internal` can safely subtract from the safety distance (a cheap but inaccurate lower bound) let's do that, and as the "flag" we use all negative numbers as "safety needs updating". So initialization and boundary crossing should set the safety to zero, the safety should only be calculated when needed (and cache the value) analogously to `find_next_step()`, and if it's an efficient shortcut we can subtract from the safety as we move in a straight line -- or we can just invalidate the distance at every internal move, and recalculate on the subsequent `find_safety()` call.

Reply:
Maybe I'm misunderstanding: I thought that photons wouldn't *ever* need the safety distance? Isn't it just for MSC and field propagation?

Reply:
Well, we can still use the safety value for photon propagation (i.e, inside LinearPropagator) if it is cached, for an example, 
```
CELER_FUNCTION
LinearPropagator::result_type LinearPropagator::operator()(real_type dist)
{
    CELER_EXPECT(dist > 0);

    result_type result;
    result.boundary = true;

    if(track_.safety() > 0 && dist < track_.safety())
    {
        result.boundary = false;
        result.distance = dist;
        track_.move_internal(dist);
    }
    else 
    {
        result.distance = track_.find_next_step();
        if (dist >= result.distance)
        {
            track_.move_to_boundary();
        }
        else
        {
            result.boundary = false;
            result.distance = dist;
            track_.move_internal(dist);
        }
    }
    return result;
}

```

Reply:
If you are not convinced or not comfortable, let's just add only the method, safety(const Real& pos) for now.

Reply:
OK, I hadn't considered that use case for photons: it would indeed be useful in diffusive (highly scattering, optically thick) regions, although I don't know whether that's seen in LHC geometry.

I'm not at all opposed to storing and caching the safety distance, I just want it to be a "pay for what you use" situation: if we're not using the safety (due to absence of field or MSC, photons, whatever) we shouldn't be constantly recalculating it. Can I push a "middle ground" to your branch to see if I'm missing something?

Reply:
I reverted the earlier commits except leaving the utility method, which should not affect anything for now - for MSC, it should pay the same price between using the cached value and recalculating, if the safety is not used elsewhere.  We may gradually integrate other user cases as necessary.

Reply:
OK, fair enough!